### PR TITLE
feat: return the account type with GetMySelf, GetUserById and GetUserByEmail and adapt tests

### PR DESCRIPTION
**Pull request by galvagnimatteo**
- Created UserType enum
- Add userType in UserInfo and UserMyself, extracting its value from mailbox's zimbraIsExternalVirtualAccount (if true account type is GUEST, INTERNAL otherwise)
- Updated ITs and UTs

Refs: UM-30

### CODE REVIEW

File: core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    // Given
+    SoapHttpUtils soapHttpUtils = simulator.getSoapHttpUtils();
+    MockServerClient mailboxServiceMock = simulator.getMailboxServiceMock();
+
+    mailboxServiceMock
+        .when(
+            HttpRequest.request()
+                .withMethod(HttpMethod.POST.toString())
+                .withPath("/service/soap/")
+                .withBody(
```
Comment (federicorispo) [inline core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java:57]: Try to add withCookie definition and see if it is neccessary

If it is necessary, change it also in R106, R157, R210

File: core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                    soapHttpUtils.getAccountInfoRequest(
+                        "fake-token", "a28fdb4d-9f4b-4c7f-a572-43cef33f1d8b")))
```
Comment (federicorispo) [inline core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java]: Pass the email istead of the account id

File: core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            HttpResponse.response()
+                .withStatusCode(HttpStatus.OK_200)
+                .withBody(
+                    soapHttpUtils.getAccountInfoResponse(
+                        "a28fdb4d-9f4b-4c7f-a572-43cef33f1d8b",
+                        "fake@example.com",
+                        "example.com",
+                        "Fake Account",
+                        "active",
+                        "FALSE")));
```
Comment (federicorispo) [inline core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java]: Pick one of these test and set this boolean to TRUE

**Review by federicorispo — DISMISSED** [review]

File: core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java (hunk 1 above)
Comment (galvagnimatteo) [reply core/src/test/java-it/com/zextras/carbonio/user_management/apis/GetUserInfoApiIT.java:57]: Cookie is not necessary

**Review by federicorispo — DISMISSED** [review]

**Review by federicorispo — DISMISSED** [review]

**Review by federicorispo — APPROVED** [review]

**Pull request merged**